docs: Add white paper index to the documentation

paulgessinger opened the pull request.

felix-russo reviewed (commented):
Should we require a doi for the repos? 
I will try and go through the workflow now

felix-russo reviewed (commented):
Really nice! Here are some things we realized during testing

Also:
We could add a folder for images and for macros to he tlatex template

nitpick: I don't know how much of a pain it is to change this, but I think there should be a space between white and paper i.e. `white-paper` instead of `whitepaper`

paulgessinger commented on whitepaper/update.py:
```diff
@@ -0,0 +1,236 @@
+
+@app.command()
+@coro
+async def pull(
```
Depends on whether it's preserved when it's manually added. I don't think there's much we can do since the file content is generated by the toml library, and as far as I know it's not possible to change the output.

paulgessinger commented on whitepaper/update.py:
```diff
@@ -0,0 +1,236 @@
+
+                status.update(f"\[{repo}] Getting metadata...")
+
+                metadata, abstract = await asyncio.gather(
+                    get_file(gh, repo, "metadata.tex"),
+                    get_file(gh, repo, "abstract.tex"),
+                )
```
Can do but like I mentioned I don't think it's actually needed.

felix-russo replied:
Actually, after we encountered this problem, I encountered it a second time... I also don't understand why it makes a difference, but the separate `await` might be a safer choice

paulgessinger replied:
ok

paulgessinger commented on docs/whitepapers/how-to-add.md:
````diff
@@ -0,0 +1,191 @@
+$ whitepaper/update.py pull
+```
+
+which will for each whitepaper listed in `whitepapers.toml`
````
Could add that in the future, for now I'd probably stick to always updating everything. I don't expect authors to have to do this regularly anyway.

paulgessinger commented on docs/tracking.md:
```diff
@@ -130,7 +130,7 @@ called *particle propagation* or *extrapolation*, is used to predict a
 the projected intersection with various types of surfaces is desired. The
 trajectory of a charged particle is governed by the [magnetic
-field](core/magnetic_field.rst) through which it travels, as well as any
+field](core/magnetic_field.md) through which it travels, as well as any
```
Yes but it won't compile locally for me otherwise. Not sure why this is not caught.

paulgessinger commented on docs/whitepapers/how-to-add.md:
```diff
@@ -0,0 +1,191 @@
+
+You also need the `convert` executable from
+[ImageMagick](https://imagemagick.org/) available on your `$PATH`.  You can
+then run 
```
I think this is out of scope for us here.

paulgessinger commented:
hey @felix-russo

I renamed every thing to use spaces, I also rolled the separate folder `whitespaces` into the `docs` folder. 
The `render` step is now executes by the documentation build automatically, so I remove that part of the documentation as well.

I also updated the generation to try to grab a release from the GitHub repo and if it finds one checks for a PDF attached to it. If found, it will link to it. I changed the template CI to run create releases when tags are created.

felix-russo reviewed (approved):
🚀
